docker.go

```diff
@@ -598,34 +600,46 @@ func (p *DockerProvider) CreateContainer(ctx context.Context, req ContainerReque
 	}
 
 	var tag string
+
+	var customOS, customArch string
 	if req.ShouldBuildImage() {
 		tag, err = p.BuildImage(ctx, &req)
 		if err != nil {
 			return nil, err
 		}
 	} else {
 		tag = req.Image
+
+		if len(req.ImagePlatform) != 0 {
+			customOS = strings.Split(req.ImagePlatform, "/")[0]
+			customArch = strings.Split(req.ImagePlatform, "/")[1]
```

Comment:
This is correct, as all platforms are formed by a string in this format: "os/arch"

👍

Reply:
A platform may also have 3 parts (os/arch/variant) in some cases. It's probably safest to use containerd's [`platforms.Parse`](https://pkg.go.dev/github.com/containerd/containerd/platforms#Parse) func for this.